src/methods/bools.jl

```diff
@@ -193,6 +171,26 @@ function point_on_line(point, line; ignore_end_vertices::Bool=false)::Bool
     return false
 end
 
+function point_on_seg(point, start, stop)
```

Comment:
I added this because the `point_on_segment` function was complicated with all of the potential endpoint options. Do we want all of those options? I removed the similar option `meets` as we discussed since it wasn't something that is available in LibGEOS. I can see how this might be helpful in other functions though...